block-storage/src/main/scala/coop/rchain/blockstorage/dag/codecs.scala

```diff
@@ -20,11 +21,26 @@ object codecs {
 
   val codecBlockHash = xmapToByteString(bytes(BlockHash.Length))
 
+  val codecApprovedBlockHash = xmapToByteString(bytes(1)) // Length of approvedBlockKey is just 1 byte
+
   val codecBlockMetadata = variableSizeBytes(uint16, bytes).xmap[BlockMetadata](
     byteVector => BlockMetadata.fromBytes(byteVector.toArray),
     blockMetadata => ByteVector(blockMetadata.toByteString.toByteArray)
   )
 
+  val codecBlockMessage = variableSizeBytesLong(uint32, bytes).xmap[BlockMessage](
+    byteVector => bytesToBlockMessage(byteVector.toArray),
+    blockMessage => ByteVector(blockMessageToBytes(blockMessage))
+  )
```

Comment:
You can use `bytes` scodec directly. Variable size is only needed when combining raw _bytes_ in a list.

The same for approved block codec.

Reply:
@tgrospic, do we have functions to convert ByteVector to ApprovedBlock and back?

Reply:
BlockMessaage and ApprovedBlock are defined with protobuf so the only correct way is to use proto variants with protobuf encoding/decoding via ByteString.
Scodec definition is just transparent wrapper which uses binary directly from protobuf, that's the reason why `bytes` scodec is enough without size prefix (from `variableSizeBytesLong` scodec).

Reply:
Probably [resolved](https://github.com/rchain/rchain/pull/3655/commits/9b9c0bded63f00fea1ea6363ab798c693d223f7a).